analytical_engine/apps/assortativity/average_degree_connectivity/average_degree_connectivity.h

```diff
@@ -0,0 +1,248 @@
+              static_cast<double>(weight * target_degree);
+        } else {
+          ctx.degree_connectivity_map[source_degree].first +=
+              static_cast<double>(weight * target_degree);
+        }
+      }
+      ctx.merge_stage = true;
+      if (frag.fid() != 0) {
+        messages.SendToFragment<degree_connectivity_t>(
+            0, ctx.degree_connectivity_map);
+      }
+      messages.ForceContinue();
+    } else {
+      // merge in worker 0
+      if (frag.fid() == 0) {
+        std::unordered_map<int, std::pair<double, double>> msg;
+        while (messages.GetMessage(msg)) {
+          for (auto& a : msg) {
+            // merge
+            if (ctx.degree_connectivity_map.count(a.first) != 0) {
+              ctx.degree_connectivity_map[a.first].first += a.second.first;
+              ctx.degree_connectivity_map[a.first].second += a.second.second;
+            } else {
+              ctx.degree_connectivity_map[a.first].first = a.second.first;
+              ctx.degree_connectivity_map[a.first].second = a.second.second;
+            }
+          }
+        }
+
+        // write to ctx
+        size_t row_num = ctx.degree_connectivity_map.size();
+        std::vector<size_t> shape{row_num, 2};
+        std::vector<double> data;
+        for (auto& a : ctx.degree_connectivity_map) {
+          double result = a.second.second == 0.0
+                              ? a.second.first
+                              : a.second.first / a.second.second;
+          ctx.degree_connectivity_map[a.first].first = result;
+          // degree
+          data.push_back(static_cast<double>(a.first));
+          // degree connectivity
+          data.push_back(result);
+        }
+        ctx.assign(data, shape);
+      }
+    }
+  }
+
+ private:
+  /**
+   * @brief traverse the neighbors of vertex v
+   *
+   * @param v
+   * @param frag
+   * @param ctx
+   * @param messages
+   */
+  void traverseVertex(vertex_t v, const fragment_t& frag, context_t& ctx,
+                      message_manager_t& messages) {
+    int source_degree =
+        getDegreeByType(frag, v, ctx.source_degree_type_, ctx.directed);
+    double norm = getWeightedDegree(v, frag, ctx);
+    if (ctx.degree_connectivity_map.count(source_degree) == 0) {
+      ctx.degree_connectivity_map[source_degree].second = norm;
+    } else {
+      ctx.degree_connectivity_map[source_degree].second += norm;
+    }
+    // process incoming neighbours
+    if (ctx.directed && ctx.source_degree_type_ == DegreeType::IN) {
+      auto oes = frag.GetIncomingAdjList(v);
+      for (auto& e : oes) {
+        edgeProcess(e, source_degree, frag, ctx, messages);
+      }
+    } else {  // process outgoing neighbours
+      auto oes = frag.GetOutgoingAdjList(v);
+      for (auto& e : oes) {
+        edgeProcess(e, source_degree, frag, ctx, messages);
+      }
+    }
+  }
+
+  template <typename T>
+  void edgeProcess(const T& e, int source_degree, const fragment_t& frag,
```

Comment:
does the process compute the neighbors degree? you can add some comment to explain what the function do?